wordpress-mobile/WordPress-Android
# Site creation rework: theme loading retry when connection resumes

Listen for network connectivity and try the themes network fetch again when connectivity is restored.

To test #1:
1. Disable wifi/cellular so the device/emulator has no connectivity
2. With the app logged in to WPCOM, add a new WPCOM site via the Site Picker
3. Tap on any of the 3 site category options in the wizard
4. Notice the "A network error occurred. Please check your connection and try again." error message.
5. Without leaving the app, enable connectivity
6. Notice the app screen showing a progress bar when connectivity restores and then loading the themes


This PR also includes the fix for #7221 . Shows a "Close" button when the site-creation process finishes but the new site has been created remotely. If it hasn't, a "Back" button is displayed.

To test #2:
* With the app logged in to wpcom, go to the site picker and start creating a new site. Bring the wizard up to step 4 (subdomain selection) but wait.
* Turn connectivity off
* Push the "CREATE SITE" button
* The "Creating" screen comes up, it informs that something went wrong and it displays a "Back" button on the toolbar
* Tap the "Back" button and notice the wizard returning to step 4

To test #3:
* With the app logged in to wpcom, go to the site picker and start creating a new site. Bring the wizard up to step 4 (subdomain selection).
* Be ready to turn connectivity off after the "Creating" screen comes up
* Push the "CREATE SITE" button and wait for the process to reach the "Retrieving site information" step
* Turn connectivity off
* The app informs that something went wrong and it displays a "X" button on the toolbar
* Tap the "X" button and notice the app returning directly to the site picker
